lib/improver/spotdata/site_data.py

```diff
@@ -0,0 +1,302 @@
+            if np.isnan(self.utc_offsets[i_site]):
+                self.utc_offsets[i_site], = set_utc_offset([site['longitude']])
+
+        return self.construct_site_dictionary()
+
+    def runtime_list(self, site_properties):
+        """
+        Use data provided on the command line/controlling suite at runtime.
+
+        Args:
+        -----
+        site_properties : dict
+            Contains:
+
+            latitudes/longitudes : lists of floats
+                Lists of coordinates at which to extract data, these define the
+                positions of the SpotData sites.
+
+            altitudes : list of floats
+                List of altitudes that can optionally be used in defining sites
+                on the fly. If unset the altitudes will be assumed as
+                equivalent to which ever neighbouring grid point is used as a
+                data source.
+
+            site_ids : list of integers
+                Optional integer site_ids that will be used as keys for the
+                site information in the returned site data dictionary. If unset
+                a simple counting index is used.
+
+        Returns:
+        --------
+        sites : dict
+            Restructured dictionary of site data.
+
+        Raises:
+        -------
+        ValueError : If runtime defined lists of site properties are of unequal
+        length, suggesting an error with the input (e.g. a site is missing
+        some required property).
+
+        """
+        n_sites = site_properties['latitudes'].__len__()
+
+        if site_properties['longitudes'].__len__() != n_sites:
+            raise ValueError('Unequal number of latitude and longitudes '
+                             'given for sites.')
+        if (('altitudes' in site_properties.keys() and
+             site_properties['altitudes'] is not None and
+             site_properties['altitudes'].__len__() != n_sites) or
+            ('site_ids' in site_properties.keys() and
+             site_properties['site_ids'] is not None and
+             site_properties['site_ids'].__len__() != n_sites)):
+            raise ValueError('Unequal number of altitudes/site_ids and '
+                             'latitude/longitudes given for sites.')
+
+        if ('site_ids' in site_properties.keys() and
+                site_properties['site_ids'] is not None):
+            self.site_ids = np.array(site_properties['site_ids'])
+        else:
+            self.site_ids = np.arange(n_sites)
+        if ('altitudes' in site_properties.keys() and
+                site_properties['altitudes'] is not None):
+            self.altitudes = np.array(site_properties['altitudes'])
+        else:
+            self.altitudes = np.full(n_sites, np.nan)
+
+        self.latitudes = np.array(site_properties['latitudes'])
+        self.longitudes = np.array(site_properties['longitudes'])
+        self.utc_offsets = set_utc_offset(self.longitudes)
+
```

Comment:
This is now gone. If we want to add it to the CLI options we can, though as you point out, I'm not sure how often the CLI defined way of doing things is likely to be used.